version: '3.8'
services:
  db_chatterflow:
    container_name: db_chatterflow
    image: postgres:13-alpine
    environment:
      POSTGRES_DB: mydatabase
      POSTGRES_USER: myuser
      POSTGRES_PASSWORD: mypassword
    volumes:
      - ./create-database.sql:/docker-entrypoint-initdb.d/create-database.sql
    ports: 
      - "5432:5432"
    env_file:
      - ./.env
  web: 
    container_name: web_chatterflow
    build: .
    ports:
      - "8000:8000"
    depends_on:
      - db_chatterflow
    volumes:
      - .:/app
    command: 
      uvicorn main:app --host 0.0.0.0 --port 8000
    env_file:
      - ./.env
    deploy:
      restart_policy:
        condition: on-failure
        delay: 5s # default
        max_attempts: 5
networks:
  name: 
    external: true
